src/harp_pedal_wizard.lua

```diff
@@ -0,0 +1,1591 @@
+end
+
+if finenv.IsRGPLua == false then
+    local ui = finenv.FCUI
+    ui:AlertInfo("This script requires RGP Lua to function.", NULL)
+    local str = finale.FCString()
+    str.LuaString = "https://robertgpatterson.com/-fininfo/-rgplua/rgplua.html"
+    ui:DisplayWebURL(str)
+end
+
+local library = require("library.general_library")
+--local configuration = require("library.configuration")
+
+local config_filename = "com.harp_pedal_wizard.text"
+
+-- finalelua library functions - copied from library\configuration.lua
+local strip_leading_trailing_whitespace = function (str)
+    return str:match("^%s*(.-)%s*$") -- lua pattern magic taken from the Internet
+end
+
+local parse_parameter -- forward function declaration
+
+
+local parse_table = function(val_string)
+    local ret_table = {}
+    for element in val_string:gmatch('[^,%s]+') do  -- lua pattern magic taken from the Internet
+        local parsed_element = parse_parameter(element)
+        table.insert(ret_table, parsed_element)
+    end
+    return ret_table
+end
+
+parse_parameter = function(val_string)
+    if '"' == val_string:sub(1,1) and '"' == val_string:sub(#val_string,#val_string) then -- double-quote string
+        return string.gsub(val_string, '"(.+)"', "%1") -- lua pattern magic: "(.+)" matches all characters between two double-quote marks (no escape chars)
+    elseif "'" == val_string:sub(1,1) and "'" == val_string:sub(#val_string,#val_string) then -- single-quote string
+        return string.gsub(val_string, "'(.+)'", "%1") -- lua pattern magic: '(.+)' matches all characters between two single-quote marks (no escape chars)
+    elseif "{" == val_string:sub(1,1) and "}" == val_string:sub(#val_string,#val_string) then
+        return parse_table(string.gsub(val_string, "{(.+)}", "%1"))
+    elseif "true" == val_string then
+        return true
+    elseif "false" == val_string then
+        return false
+    end
+--    return tonumber(val_string)
+    return val_string
+end
+
+local get_parameters_from_file = function(file_name) -- modified
+    local parameters = {}
+    for line in io.lines(file_name) do
+        local comment_marker = "--"
+        local parameter_delimiter = "="
+        local comment_at = string.find(line, comment_marker, 1, true) -- true means find raw string rather than lua pattern
+        if nil ~= comment_at then
+            line = string.sub(line, 1, comment_at-1)
+        end
+        local delimiter_at = string.find(line, parameter_delimiter, 1, true)
+        if nil ~= delimiter_at then
+            local name = strip_leading_trailing_whitespace(string.sub(line, 1, delimiter_at-1))
+            local val_string = strip_leading_trailing_whitespace(string.sub(line, delimiter_at+1))
+            parameters[name] = parse_parameter(val_string)
+        end
+    end
+    return parameters
+end
+
+-- Modified from library
+function get_parameters(file_name, parameter_list)
```

Comment:
Why are you replicating the library function? Instead, if the library function doesn't do exactly what you need, enhance it to do that. Don't duplicate the code.

Reply:
It appears the only reason you've duplicated the library code is to be able to use a different file path. Instead of duplicating the code, add a function to the config library called something like `get_parameters_with_path`. Actually, the most correct steps would be:

- Change `get_parameters` to `get_parameters_with_full_path`
- Add a new `get_parameters` that calls `get_parameters_with_full_path` with the expected `script_settings` path.

This expands the library for everyone instead of being a one-off.

Reply:
As a followup, I am going to take a stab at enhancing the configuration library for user settings. So look for a pull request from me on this.